### PR TITLE
fix(docs): improve readme "why a language"

### LINKED ISSUE #2181: Improve "why a language" explanation
### Feature Spec

From Reddit: https://www.reddit.com/r/devops/comments/12n4sna/comment/jgeloz2/?utm_source=share&utm_medium=web2x&context=3 

> "I would recommend something like showing how some code would look in a hypothetical Typescript or Python version of Wing and then show the same code being smaller or better or easier in your language. Pick maybe the top 3 examples in your mind that your language is worth the learning-time investment because it's just so much better. Because otherwise that's a very big ask for a team that already knows Python or Typescript or Go."

> "I probably wouldn't compare it to the compiled output since that's not usually a fair thing, compiled output is usually not intended to be the same as human written code. But just imagine if you were to try and make something like Wing without a custom language, what would it probably end up looking like, and then show why Wing fixes some specific problem there.

> Because otherwise it's a very hard sell. Existing languages have linters and CI workflows and dependency management and a million other little things that will need to be rewritten for Wing. And just showing me some nice Wing code and saying "see, isn't it nice" doesn't really give me enough to want to take on that massive pile of work to use your project."

> "the cloud is fundamentally different from legacy computing, and legacy programming paradigms are a continuing source of friction when solutioning for the cloud. approaching these problems with a new coding paradigm allows us to provide purpose built tools that will reduce operational overhead by directly addressing the needs of the cloud practitioner.

> then some bullets that help convince me this paragraph isnt a lie, probably about conceptual features. this list could be a pared down and tweaked version of the list titled "Wing is designed to address these challenges in the following ways:" in the "new programming paradigm" click through link.

> then you can do whatever you want with the rest of the page."

### Use Cases

Developers wanting to understand why this is a language and not a library or language extension

### Implementation Notes

This should go to both the FAQ version and possibly the main readme of the project

### Component

Documentation

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by eladb**
Fixes #2181

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,38 +29,46 @@ Our mission is to bring back your creative flow and close the gap between imagin
...
+  bucket.put("file-${i}.txt", message);
+});
+```
 
-// An `inflight` represents code that runs later, on other machines,
-// and can interact with any cloud resources
-let hello_world = inflight () => {
-  bucket.put("hello.txt", "Hello, World!");
-};
+The `cloud.Bucket`, `cloud.Counter` and `cloud.Bucket` are *preflight objects*. They represent
```
Comment (ShaiBer) [inline README.md:60]:
```suggestion
The `cloud.Queue`, `cloud.Counter` and `cloud.Bucket` are *preflight objects*. They represent
```

File: README.md [hunk 2]
```diff
@@ -29,38 +29,46 @@ Our mission is to bring back your creative flow and close the gap between imagin
...
+cloud infrastructure resources. When compiled to AWS, a Terraform file will be produced with the AWS implementation
```
Comment (ShaiBer) [inline README.md:61]: I don't want it to sound like we are an AWS shop

```suggestion
cloud infrastructure resources. When compiled to a specific cloud provider, such as AWS, a Terraform file will be produced with the provider's implementation
```

**Review by ShaiBer — CHANGES_REQUESTED** [review]
I think that it's a step in the right direction and makes the readme shorter and more exact.
It doesn't answer all the feedback that arose from the Reddit post though because:
1. There is no direct comparison of the same code in Wing and in other languages/frameworks.
2. There is no clear statement of exactly what we can do with Wing that cannot be done with a library/compiler extension, and why. BTW, I think this part may be long and too technical for most, so maybe best to put as an FAQ topic and link to it

**Comment by eladb** [comment]
> There is no clear statement of exactly what we can do with Wing that cannot be done with a library/compiler extension

I thought this is that statement:

"In existing languages, where there is no way to distinguish between multiple execution phases, it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific
execution phase."

No?

**Comment by ShaiBer** [comment]
> I thought this is that statement:
> 
> "In existing languages, where there is no way to distinguish between multiple execution phases, it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific execution phase."
> 
> No?

I don't think it will be clear enough to most readers. Also, it doesn't explain why it cannot be done by other means that are not a language (there are some attempts at similar things by others, like Pulumi and we don't explain where they hit a wall that we don't )

File: README.md [hunk 3]
```diff
@@ -29,38 +29,46 @@ Our mission is to bring back your creative flow and close the gap between imagin
...
-new cloud.Function(hello_world);
-```
+The `queue.add_consumer()` method is a *preflight method* that configures the infrastructure to
+invoke a particular *inflight function* for each message in the queue.
+
+Now comes the cool part: the code that runs inside the inflight function interacts with the `counter` and the `bucket` objects
+through the *inflight methods* of these objects (`counter.inc()` and `bucket.put()`). These methods can only be
+called from inflight scopes. In existing languages, where there is no way to distinguish between multiple execution phases,
+it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific
+execution phase.
```
Comment (Chriscbr) [inline README.md:71]: (Optional)
```suggestion
called from inflight scopes. In existing languages, where there is no way to distinguish between multiple execution phases,
it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific
execution phase in a type-safe manner.
```

or

```suggestion
called from inflight scopes. In existing languages, where there is no way to distinguish between multiple execution phases in the type system,
it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific
execution phase.
```

**Comment by Chriscbr** [comment]
> Also, it doesn't explain why it cannot be done by other means that are not a language (there are some attempts at similar things by others, like Pulumi and we don't explain where they hit a wall that we don't )

Maybe we can include a nod to how execution phases as a first-class citizen/concept means your IaC code becomes more composable and reusable, or how it enables resource to infer IAM policies automatically? I could also see some of this being left for the tutorial or getting started guide

**Comment by ShaiBer** [comment]
> > There is no clear statement of exactly what we can do with Wing that cannot be done with a library/compiler extension
> 
> I thought this is that statement:
> 
> "In existing languages, where there is no way to distinguish between multiple execution phases, it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific execution phase."
> 
> No?

I like it

**Review by ShaiBer — APPROVED** [review]
Approve, and would like to see your suggestion for making the message even more clear added later as well

**Pull request merged**